compiler/rustc_middle/src/middle/resolve_lifetime.rs

```diff
@@ -83,4 +100,6 @@ pub struct ResolveLifetimes {
     /// to the trait object lifetime defaults computed from them.
     pub object_lifetime_defaults:
         FxHashMap<LocalDefId, FxHashMap<ItemLocalId, Vec<ObjectLifetimeDefault>>>,
+
+    pub scope_for_path: FxHashMap<LocalDefId, FxHashMap<ItemLocalId, LifetimeScopeForPath>>,
```

Comment:
Because of this change, I would like to run a perf run before merging to look at the memory consumption change.

Reply:
Yes that's a good idea.  Recursively collecting all lifetimes in scope might be a little expensive computationally. We could also just restrict this to elision information, since that's more helpful for the diagnostics. Can your request the perf run? I don't think I have the priviliges to request one.

Reply:
Lets fix the conflict and rebase and then I'll start the perf run.

Reply:
Rebased.